### PR TITLE
Fix Animation Preview height & being stuck when duplicating an image

### LINKED ISSUE #4461: Regressions in the animation previews
## Describe the bug

- The preview is not taking the whole height of the window.
- The default zoom seems also broken (maybe linked to the previous bug)
- The loading should probably not be shown at all here.
- The looping seems also broken 

image

## To Reproduce

- Open the preview animation window.
- For the looping issue:
  image

## Other details

* Version 5.1.149, desktop

### PR DESCRIPTION
**Pull request by ClementPasteau**
Fixes #4461

### CODE REVIEW

File: newIDE/app/src/ObjectEditor/Editors/SpriteEditor/AnimationPreview.js [hunk 1]
```diff
@@ -161,17 +164,28 @@ const AnimationPreview = ({
...
         if (currentFrameIndex !== newFrameIndex) {
-          imagesLoadedArray.current[currentFrameIndexRef.current] = false;
-          // When the array of loaders changes, wait a bit to display the loader to avoid flickering.
-          loaderTimeout.current = setTimeout(() => {
-            console.warn(
-              'The image took too long to load, displaying a loader.'
-            );
-            setIsStillLoadingResources(true);
-          }, 500);
+          if (newResourceName === currentResourceNameRef.current) {
```
Comment (AlexandreSi) [inline newIDE/app/src/ObjectEditor/Editors/SpriteEditor/AnimationPreview.js:171]: Shouldn't it be the path rather than the name of the file?

File: newIDE/app/src/ObjectEditor/Editors/SpriteEditor/AnimationPreview.js [hunk 2]
```diff
@@ -221,7 +235,7 @@ const AnimationPreview = ({
 
   return (
     <Column expand noOverflowParent noMargin>
-      <div style={styles.imageContainer}>
```
Comment (AlexandreSi) [inline newIDE/app/src/ObjectEditor/Editors/SpriteEditor/AnimationPreview.js:224]: You can remove the style I think

File: newIDE/app/src/ObjectEditor/Editors/SpriteEditor/DirectionTools.js [hunk 3]
```diff
@@ -185,6 +185,7 @@ export default class DirectionTools extends Component<Props, State> {
                 this.setState({ timeBetweenFrames: text })
               }
               isLooping={direction.isLooping()}
+              hideAnimationLoader // No need to show a loader in the Direction Tools.
```
Comment (AlexandreSi) [inline newIDE/app/src/ObjectEditor/Editors/SpriteEditor/DirectionTools.js:188]: It is true when on Desktop because the resources are local, but should we show the loader for cloud projects?

File: newIDE/app/src/ObjectEditor/Editors/SpriteEditor/AnimationPreview.js (hunk 1 above)
Comment (ClementPasteau) [reply newIDE/app/src/ObjectEditor/Editors/SpriteEditor/AnimationPreview.js:171]: Tried that, but:
- there is a one to one relation between name & path (and a function is called to get the path from the name) so the result should be the same
- but when a file is cached locally, it adds a `?cache={date}` in the url, so the urls actually differ as the animation continues, but the browser is smart enough to used the cached image and not trigger the onLoad.

tldr; I tried but it felt more complex to make it work

File: newIDE/app/src/ObjectEditor/Editors/SpriteEditor/DirectionTools.js (hunk 3 above)
Comment (ClementPasteau) [reply newIDE/app/src/ObjectEditor/Editors/SpriteEditor/DirectionTools.js:188]: We could, but as this tool is not the "external facing" animations, like in the Asset Store, and more of a "internal tool" to work on your animation, I'm not sure it adds value.

File: newIDE/app/src/ObjectEditor/Editors/SpriteEditor/AnimationPreview.js (hunk 1 above)
Comment (AlexandreSi) [reply newIDE/app/src/ObjectEditor/Editors/SpriteEditor/AnimationPreview.js:168]: Are you sure there is a one to one relation between names and paths for resources? Because today you can have multiple objects using the same resource.

Comment (ClementPasteau) [reply newIDE/app/src/ObjectEditor/Editors/SpriteEditor/AnimationPreview.js:168]: I think this is different here, this is in the scope of 1 object & 1 animation only.
The animation knows where to find the resource from the name of this resource.
It uses the function `resourcesLoader.getResourceFullUrl(project, name, {})` in the `DirectionTools`

**Review by AlexandreSi — APPROVED** [review]
LGTM
